src/models/Model.ts

```diff
@@ -94,11 +94,11 @@ export class Model extends BaseComponent {
    * @returns {{object}}
    */
   public getAttributes() {
-    let attributes: { [key: string]: any } = {};
+    const attributes: { [key: string]: any } = {};
     _.each(this.attributes, (attribute, key) => {
       if (_.isObject(attribute)) {
         if (!Utils.objectEqual(attribute, this.defaultAttributes[key])) {
-          attributes[key] = attribute;
+          attributes[key] = Utils.extendDeep({}, attribute);
```

Comment:
If you want to copy an object, I prefer to use object spread `attributes[key] = {...attributes}`

Reply:
Changing this one to `attributes[key] = {...attribute}` fails some tests:

```
HistoryController should set the query state model representation on all event in the hash

Expected '#c=notDefault&d={"0":1 , "1":2 , "2":3}' to be '#c=notDefault&d=[1,2,3]'.
```